imap_processing/mag/l0/mag_l0_data.py

```diff
@@ -99,9 +101,17 @@ def __post_init__(self):
         """
         if isinstance(self.VECTORS, str):
             # Convert string output from space_packet_parser to bytearray
-            self.VECTORS = bytearray(
-                int(self.VECTORS, 2).to_bytes(len(self.VECTORS) // 8, "big")
+            self.VECTORS = np.frombuffer(
+                int(self.VECTORS, 2).to_bytes(len(self.VECTORS) // 8, "big"),
+                dtype=np.dtype(">b"),
             )
 
+        if isinstance(self.VECTORS, bytearray):
+            self.VECTORS = np.array(self.VECTORS, dtype=np.dtype(">b"))
```

Comment:
I had both cases happen during testing (I assume due to space_packet_parser) but let me double check this. It could be that I had both cases happen because I had already converted VECTORS to a bytearray or something.